### PR TITLE
Global running

### LINKED ISSUE #282: [8pt] Start reworks
I don't like some `start,stop,status` (I will call that's all `running` commands)  features in the old CLI, so I suggest to change it in a new version (since it's a next major version that can be incompatible).

## Intro

In the old cli running commands had these options (with default values):

* `--run-dir` - directory where pid and sock files are stored (`/var/run/tarantool`);
* `--apps-path` - path to apps directory when running in multi-app env (`/usr/share/tarantool`).
   App directory is `<apps-path>/<app-name>`;
* `--cfg` -  cartridge instances config file (`instances.yml`);
* `--stateboard`, `--stateboard-only` flags indicates if stateboard instance should be managed with or instead of app instances ;
* `--script` - application entrypoint (`init.lua`).

All these options has default values that can be overwritten  by `TARANTOOL_*` env variables or values from `./.cartridge.yml` or `~/.cartridge.yml`.

New version of CLI doesn't support `apps-path` option.

### Problem

Application can be started from it's directory (local) or from arbitrary place (system).
In the system case we discover application as `<apps-path>/<app-name>`.

Now, all default options except `--cfg` are defined for system start. 
If current directory contains rockspec than current application is running, else
cli is looking for it in `/usr/share/tarantool/<app-name>`. 
In this case, if app dir doesn't contain rockspec, you get error that your app isn't
found in `/usr/share/tarantool/<app-name>` and it's a bit confused.

The second case is a `cfg` file - now it's `./instances.yml` in the current directory.
It's OK for local running, but if you wand to manage apps from any place, it leads to frustration...
For system running we need to use `/etc/tarantool/conf.d` as a default value. 

The same about `.cartridge.yml` file.

### Solution

I sure that user should specify if he wants `system` running or `local`.

I suggest to introduce `--system` flag, that is `false` by default.
Local running is used *only for running from app directory*.

System running requires app name specifying. 
(No `start router`, only `start --name myapp router` or `start myapp router s1-master`)

Since CLI can start two or more instances (`cartridge start router s1-master ...`), 
we should deny to specify different apps instances in one command.

Depends on mode, options has different default values:

**run-dir**:
  * local: `./tmp/run`
  * system: `/var/run/tarantool`

**cfg**:
  * local: `./instances.yml`
  * system: `/etc/tarantool/conf.d`

**apps-path**:
  * local: Is not allowed - local running it's for starting one app in it's directory
  * system: `/usr/share/tarantool`

**data-dir**: (in a new CLI)
  * local: `./tmp/data`
  * system: `/var/lib/tarantool`

**log-dir**: (in a new CLI)
  * local: `./tmp/log`
  * system: `/var/log/tarantool`

**.cartridge,yml** (contains default values for all opts above)
  * local: `./.cartridge,yml`
  * system: `~/.cartridge,yml`

**script**
  * local: `./init.lua`
  * system: `<apps-path>/<app-name>/init.lua`

### PR DESCRIPTION
**Pull request by dokshina**
Closes #282

### CODE REVIEW

File: cli/commands/usage.go [hunk 1]
```diff
@@ -127,7 +140,7 @@ Ignored if "--stateboard-only" is specified
 `
 
 	stateboardOnlyUsage = `Manage only application stateboard
-If specified, "INSTANCE_NAME..." are ignored.
```
Comment (olegrok) [inline cli/commands/usage.go:130]: Why did you do it?

File: cli/project/files.go [hunk 2]
```diff
@@ -5,36 +5,51 @@ import (
...
 	"path/filepath"
 
+	"github.com/apex/log"
 	"github.com/tarantool/cartridge-cli/cli/common"
 	"github.com/tarantool/cartridge-cli/cli/context"
 )
 
 const (
-	cartridgeLocalConf = ".cartridge.yml"
+	defaultHomeDir = "/home"
```
Comment (olegrok) [inline cli/project/files.go:14]: Why not `$HOME`?

Comment (olegrok) [reply cli/project/files.go:14]: See here - https://github.com/tarantool/cartridge-cli/pull/315/files#diff-660f4520ad6b4abeaabb003310e50b43R177

But you don't check access rights for this directory

File: cli/project/files.go [hunk 3]
```diff
@@ -152,47 +167,114 @@ func getPath(conf map[string]interface{}, opts PathOpts) (string, error) {
...
+// * ./.cartridge.yml for local running
+// * ~/.cartridge.yml for global running
 //
 // The priority of sources is:
 // * user-specified flags
-// * value from .cartridge.yml
+// * value from .cartridge.yml (if useConf is true)
 // * default values (defined here in const section)
-func SetLocalRunningPaths(ctx *context.Ctx) error {
+func SetRunningPaths(ctx *context.Ctx, useConf bool) error {
```
Comment (olegrok) [inline cli/project/files.go:223]: Seems it could be splitted to `SetLocalRunningPaths ` and `SetGlobalRunningPaths `. If here is quite big

Comment (olegrok) [reply cli/project/files.go:223]: Common part could be moved into separate function

Comment (dokshina) [reply cli/project/files.go:223]: `useConf ` variable isn't about local/global, it's about running/pack (in pack rpm and deb the same global paths are used).

The main difference of `SetLocalRunningPaths` and `SetGlobalRunningPaths` is the default values and `App(s)Dir` variable. And I sure that in the one functions it's quite simple to understand the difference and support this code.

File: cli/commands/usage.go (hunk 1 above)
Comment (dokshina) [reply cli/commands/usage.go:130]: Because I decided to remove trailing dots.

Comment (olegrok) [reply cli/commands/usage.go:130]: Is it only one place?

Could you please move such refactoring to separate patch if there are other places?

Comment (dokshina) [reply cli/commands/usage.go:130]: I've done it in the previous patch, but missed this place.

File: cli/project/files.go (hunk 2 above)
Comment (dokshina) [reply cli/project/files.go]: Fixed:

* `os.UserHomeDir()` is used now (BTW, it returns `${HOME}`);
* if getting home directory fails, cli returns an error.

File: cli/commands/usage.go [hunk 4]
```diff
@@ -84,14 +84,27 @@ result artifact
...
+	runningCommonUsage = `Manage instance(s) of current application
 
-Application name is described from rockspec in the current directory.
+There are two modes of running instances: local and global.
+If --global flag is specified, then application from <apps-dir>/<name>
+is started and global paths (run-dir, data-dir, log-dir, cfg)
+are used, otherwise application from the current directory is started
+and all application directories and files are placed in the current directory.
+
+I case of local running, application name is taken from rockspec in
```
Comment (olegrok) [inline cli/commands/usage.go:95]: typo: `In case`

Comment (olegrok) [reply cli/commands/usage.go:95]: `rockspec in` -> `rockspec from current directory`?

File: cli/commands/usage.go [hunk 5]
```diff
@@ -84,14 +84,27 @@ result artifact
...
+the current directory. For global running name should be passed using
+--name option.
+
+Some flags default options can be overridden in .cartridge.yml config file.
```
Comment (olegrok) [inline cli/commands/usage.go:99]: `config file` -> `configuration file`?

**Comment by dokshina** [comment]
We've decided that this feature is unused.

**Pull request closed without merging**